Bottom: e36d9dc3b3eb324408c63f19f95cdb5feb1eaaf2
Top:    ddae8b1503f6af578f6f774c2b7c3895fb134021
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-03 14:50:06 +0100

hid/gtk: Don't use an hbox to pack the board name label

The hbox only has one child, so use the name label directly.


---

diff --git a/src/hid/gtk/gui-top-window.c b/src/hid/gtk/gui-top-window.c
index 651f7f5..283b345 100644
--- a/src/hid/gtk/gui-top-window.c
+++ b/src/hid/gtk/gui-top-window.c
@@ -636,13 +636,13 @@ ghid_window_set_name_label (gchar * name)
 
   if (ghidgui->ghid_title_window)
     {
-      gtk_widget_hide (ghidgui->label_hbox);
+      gtk_widget_hide (ghidgui->name_label);
       str = g_strdup_printf ("%s%s (%s) - PCB", PCB->Changed ? "*": "",
                              ghidgui->name_label_string, filename);
     }
   else
     {
-      gtk_widget_show (ghidgui->label_hbox);
+      gtk_widget_show (ghidgui->name_label);
       str = g_strdup_printf (" <b><big>%s</big></b> ",
                              ghidgui->name_label_string);
       gtk_label_set_markup (GTK_LABEL (ghidgui->name_label), str);
@@ -1271,20 +1271,15 @@ ghid_build_pcb_top_window (void)
    * The board name is optionally in compact_vbox and the position
    * labels will be packed below or to the side.
    */
-  ghidgui->label_hbox = gtk_hbox_new (FALSE, 0);
-  gtk_box_pack_start (GTK_BOX (ghidgui->compact_vbox), ghidgui->label_hbox, TRUE, FALSE, 2);
-
-  label = gtk_label_new ("");
-  gtk_label_set_use_markup (GTK_LABEL (label), TRUE);
+  ghidgui->name_label = gtk_label_new ("");
+  gtk_label_set_use_markup (GTK_LABEL (ghidgui->name_label), TRUE);
   if (ghidgui->name_label_string)
-    s =
-      g_strdup_printf (" <b><big>%s</big></b> ", ghidgui->name_label_string);
+    s = g_strdup_printf (" <b><big>%s</big></b> ", ghidgui->name_label_string);
   else
     s = g_strdup ("<b><big>%s</big></b>");
-  gtk_label_set_markup (GTK_LABEL (label), s);
+  gtk_label_set_markup (GTK_LABEL (ghidgui->name_label), s);
   g_free (s);
-  gtk_box_pack_start (GTK_BOX (ghidgui->label_hbox), label, FALSE, TRUE, 4);
-  ghidgui->name_label = label;
+  gtk_box_pack_start (GTK_BOX (ghidgui->compact_vbox), ghidgui->name_label, TRUE, FALSE, 6);
 
   /*
    * The position_box pack location depends on user setting of
diff --git a/src/hid/gtk/gui.h b/src/hid/gtk/gui.h
index e7e5f55..8dfc203 100644
--- a/src/hid/gtk/gui.h
+++ b/src/hid/gtk/gui.h
@@ -104,7 +104,7 @@ typedef struct
   GtkEntry *command_entry;
 
   GtkWidget *top_hbox,
-    *menu_hbox, *compact_vbox, *compact_hbox, *position_hbox, *label_hbox,
+    *menu_hbox, *compact_vbox, *compact_hbox, *position_hbox,
     *mode_buttons0_vbox, *mode_buttons1_hbox, *mode_buttons1_vbox,
     *mode_buttons0_frame, *mode_buttons1_frame, *mode_buttons0_frame_vbox;
   GtkWidget *menu_bar, *layer_selector;
